server/app/workspaces.go

```diff
@@ -20,12 +20,12 @@ func (a *App) GetRootWorkspace() (*model.Workspace, error) {
 		}
 		err := a.store.UpsertWorkspaceSignupToken(*workspace)
 		if err != nil {
-			a.logger.Fatal("Unable to initialize workspace", mlog.Err(err))
```

Comment:
This does more than change the log level.  Fatal exited the app with an error code as this condition was considered unrecoverable, and this behaviour would be changed by switching to `Error`.  Is this intentional?

Reply:
Wherever GetRootWorkspace method is referred in codebase, there is logic to check error is nil, Since Fatal was terminating the application, the logic on the other modules are not useful, so made it as error. and added test cases for the same

below is the code sample from server.go
`      // Init workspace
	if _, err := app.GetRootWorkspace(); err != nil {
		params.Logger.Error("Unable to get root workspace", mlog.Err(err))
		return nil, err
	}`

So, that's why made this change and added the testcase

Reply:
@wiggin77 Contributor has responded